lib/async_hooks.js

```diff
@@ -155,10 +155,24 @@ class AsyncResource {
       throw new ERR_INVALID_ASYNC_ID('triggerAsyncId', triggerAsyncId);
     }
 
-    this[async_id_symbol] = newAsyncId();
-    this[trigger_async_id_symbol] = triggerAsyncId;
-    // this prop name (destroyed) has to be synchronized with C++
-    this[destroyedSymbol] = { destroyed: false };
+    Object.defineProperties(this, {
+      [async_id_symbol]: {
+        value: newAsyncId(),
+        configurable: true,
+        writable: true
+      },
+      [trigger_async_id_symbol]: {
+        value: triggerAsyncId,
+        configurable: true,
+        writable: true
+      },
```

Comment:
these should be kept enumerable as they are super-useful for debugging.

Reply:
It is still possible to look at those when using the `showHidden` option as in: `console.log(util.inspect(obj, { showHidden: true }))`.

Or do you think it should be available for regular users as well?

Reply:
> Or do you think it should be available for regular users as well?

I would say so. Also, these objects shouldn’t show up as part of debugging another Node core object by accident, so inspecting them is probably always intended to show the state of this particular hook.